secure-api-gateway-ob-uk-rs-server/src/main/java/com/forgerock/sapi/gateway/ob/uk/rs/server/api/obie/payment/v4_0_0/vrp/DomesticVrpsApiController.java

```diff
@@ -0,0 +1,288 @@
+
+        OBDomesticVRPDetailsDataPaymentStatusInnerStatusDetailStatusReason statusReasonEnum = OBDomesticVRPDetailsDataPaymentStatusInnerStatusDetailStatusReason.PENDINGSETTLEMENT;
+        String localInstrument = paymentSubmission.getPayment().getData().getInstruction().getLocalInstrument();
+        OBDomesticVRPDetails vrpDetails = new OBDomesticVRPDetails()
+                .data(
+                        new OBDomesticVRPDetailsData()
+                                .addPaymentStatusItem(
+                                        new OBDomesticVRPDetailsDataPaymentStatusInner()
+                                                .status(FRPaymentDetailsStatusConverter.toOBDomesticVRPDetailsDataPaymentStatusInnerStatus(paymentSubmission.getStatus().getValue()))
+                                                .paymentTransactionId(paymentSubmission.getTransactionId())
+                                                .statusUpdateDateTime(new DateTime(paymentSubmission.getUpdated()))
+                                                .statusDetail(
+                                                        new OBDomesticVRPDetailsDataPaymentStatusInnerStatusDetail()
+                                                                .localInstrument(localInstrument)
+                                                                .status(FRPaymentDetailsStatusConverter.toOBDomesticVRPDetailsDataPaymentStatusInnerStatus(paymentSubmission.getStatus().getValue()).getValue())
+                                                                .statusReason(FRPaymentDetailsStatusConverter.toOBDomesticVRPDetailsDataPaymentStatusInnerStatus(paymentSubmission.getStatus().getValue()).getValue())
+                                                                .statusReasonDescription(statusReasonEnum.getValue())
+                                                )
+                                )
+
+                );
+        return ResponseEntity.ok(vrpDetails);
+    }
+
+    @Override
+    public ResponseEntity<OBDomesticVRPResponse> domesticVrpPost(
+            String authorization,
+            String xJwsSignature,
+            String xIdempotencyKey,
+            OBDomesticVRPRequest obDomesticVRPRequest,
+            String xFapiAuthDate,
+            String xFapiCustomerIpAddress,
+            String xFapiInteractionId,
+            String xCustomerUserAgent,
+            String xVrpLimitBreachResponseSimulation,
+            String apiClientId,
+            HttpServletRequest request,
+            Principal principal
+    ) throws OBErrorResponseException, OBErrorException {
+        log.debug("Received VRP payment submission: '{}'", obDomesticVRPRequest);
+
+        paymentSubmissionValidator.validateIdempotencyKey(xIdempotencyKey);
+
+        String consentId = obDomesticVRPRequest.getData().getConsentId();
+        final DomesticVRPConsent consent = consentStoreClient.getConsent(consentId, apiClientId);
```

Comment:
Could this be null?

Reply:
Yes, it can be null but it's not an issue because this is treated in the consentStoreClient. If a consent is not found and not found error will be thrown there.

Reply:
What about line 205 where it's accessed (if it can be null)?